pypeit/coadd1d.py

```diff
@@ -207,9 +209,101 @@ def load(self):
+        Exposures with flux masked everywhere are always removed.
+        Exposures that are considered bad based on their S/N compared to the
+        average S/N among all the exposures, are removed only if self.par['sigrej_exp'] is set.
+        The attributes self.waves, self.fluxes, self.ivars, self.gpms need to be defined.
+
+        Returns
+        -------
+        gpm_exp: list of bool
+            List of boolean that indicates which exposures
+            have been coadded. The length of the list is nexp.
+        _waves : list of float `numpy.ndarray`_
+            Updated list of wavelength arrays.
+        _fluxes : list of float `numpy.ndarray`_
+            Updated list of flux arrays.
+        _ivars : list of float `numpy.ndarray`_
+            Updated list of inverse variance arrays.
+        _gpms : list of bool `numpy.ndarray`_
+            Updated list of good pixel mask variance arrays.
+        """
+
+        # initialize the exposures lists
+        _waves = [wave for wave in self.waves]
+        _fluxes = [flux for flux in self.fluxes]
+        _ivars = [ivar for ivar in self.ivars]
+        _gpms = [gpm for gpm in self.gpms]
+        _spec1dfiles = [spec1dfile for spec1dfile in self.spec1dfiles]
+        _objids = [objid for objid in self.objids]
+
+        # good exposures index
+        goodindx_exp = np.arange(self.nexp)
+
+        # check if there are exposures that are completely masked out, i.e., gpms = False for all spectral pixels
+        masked_exps = [np.all(np.logical_not(gpm)) for gpm in _gpms]
+        if np.any(masked_exps):
+            msgs.warn(f'The following exposure(s) is/are completely masked out. It/They will not be coadded.')
+            [msgs.warn(f"Exposure {i}: {fname.split('/')[-1]}  {obj}")
+             for i, (fname, obj, masked_exp) in enumerate(zip(_spec1dfiles, _objids, masked_exps)) if masked_exp]
+            # remove masked out exposure
+            _waves = [wave for (wave, masked_exp) in zip(_waves, masked_exps) if not masked_exp]
+            _fluxes = [flux for (flux, masked_exp) in zip(_fluxes, masked_exps) if not masked_exp]
+            _ivars = [ivar for (ivar, masked_exp) in zip(_ivars, masked_exps) if not masked_exp]
+            _gpms = [gpm for (gpm, masked_exp) in zip(_gpms, masked_exps) if not masked_exp]
+            _spec1dfiles = [spec1dfile for (spec1dfile, masked_exp) in zip(_spec1dfiles, masked_exps) if not masked_exp]
+            _objids = [objid for (objid, masked_exp) in zip(_objids, masked_exps) if not masked_exp]
+            # update good exposures index
+            goodindx_exp = goodindx_exp[np.logical_not(masked_exps)]
+
+        # check if there is still more than 1 exposure left
+        if len(_fluxes) < 2:
+            msgs.error('At least 2 exposures are required for coadding.')
+
+        # check if there is any bad exposure by comparing the rms_sn with the median rms_sn among all exposures
+        if len(_fluxes) > 2:
```

Comment:
This masking based on a statistical analysis of the rms_sn needs to NOT be the default behavior. My suggestion is that you set sigrej_exp to default to None. And that you only carry out the code below if sigrej_exp is not None.

Reply:
Masking is not done if `sigrej_exp=None`. You can see below: 
```
                if self.par['sigrej_exp'] is not None:
                    msgs.warn('The above exposure(s) will not be coadded.')
```

There is still some stats on the S/N, which is printed in the terminal, when `sigrej_exp=None`. I think it's useful to have that information, but the masking is not applied.

Reply:
Okay I agree printing the information is useful. I would some text to the warning if sigrej_exp is not None that indicates the exposure is being masked. Right now the user won't know it is masked from watching the terminal outputs.